### PR TITLE
[Merged by Bors] - feat: port Data.Bool.AllAny

### PR DESCRIPTION
**Pull request by mo271**

### CODE REVIEW

File: Mathlib/Data/Bool/AllAny.lean [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+namespace List
+
+--#print List.all_nil
+/-
+@[simp]
+theorem all_nil (p : α → Bool) : all [] p = tt :=
+  rfl
+#align list.all_nil List.all_nil
+-/
```
Comment (mo271) [inline Mathlib/Data/Bool/AllAny.lean:31]: This is already in `lake-packages/std/Std/Data/List/Init/Lemmas.lean`
Should I just delete those lines then?

File: Mathlib/Data/Bool/AllAny.lean [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+/- warning: list.all_cons -> List.all_cons is a dubious translation:
+lean 3 declaration is
+  forall {α : Type.{u1}} (p : α -> Bool) (a : α) (l : List.{u1} α), Eq.{1} Bool (List.all.{u1} α (List.cons.{u1} α a l) p) (and (p a) (List.all.{u1} α l p))
+but is expected to have type
+  forall {α : Type.{u1}} {p : α} {a : List.{u1} α} {l : α -> Bool}, Eq.{1} Bool (List.all.{u1} α (List.cons.{u1} α p a) l) (and (l p) (List.all.{u1} α a l))
+Case conversion may be inaccurate. Consider using '#align list.all_cons List.all_consₓ'. -/
+@[simp]
+theorem all_consₓ (p : α → Bool) (a : α) (l : List α) : all (a :: l) p = (p a && all l p) :=
```
Comment (mo271) [inline Mathlib/Data/Bool/AllAny.lean:40]: `all_cons` and `any_cons` are also in the `lake-packages/std/Std/Data/List/Init/Lemmas.lean` and I followed the suggested to add ₓ.

File: Mathlib/Data/Bool/AllAny.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Data/Bool/AllAny.lean:31]: You should definitely keep the `#align`, even if the lemma is in Std now

Comment (mo271) [reply Mathlib/Data/Bool/AllAny.lean]: done, added a porting note saying this is now in Std.

File: Mathlib/Data/Bool/AllAny.lean (hunk 2 above)
Comment (ChrisHughes24) [reply Mathlib/Data/Bool/AllAny.lean:29]: I think you're only supposed to add the  ₓ to the align statement and not the actual lemma name

Comment (mo271) [reply Mathlib/Data/Bool/AllAny.lean]: got it, done

Comment (mo271) [reply Mathlib/Data/Bool/AllAny.lean]: but then it doesn't build...

Comment (mo271) [reply Mathlib/Data/Bool/AllAny.lean]: done

File: Mathlib/Data/Bool/AllAny.lean [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+! leanprover-community/mathlib commit 5a3e819569b0f12cbec59d740a2613018e7b8eec
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.List.Basic
+
+/-!
+# Boolean quantifiers
+
+This proves a few properties about `list.all` and `list.any`, which are the `bool` universal and
```
Comment (jcommelin) [inline Mathlib/Data/Bool/AllAny.lean:16]:
```suggestion
This proves a few properties about `List.all` and `List.any`, which are the `bool` universal and
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**